ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -353,8 +353,12 @@ void fillChunks() {
       if (chunkFillingCompletedSuccessfully) {
         PutChunk lastChunk = getBuildingChunk();
         if (lastChunk != null) {
-          lastChunk.onFillComplete(true);
-          updateChunkFillerWaitTimeMetrics();
+          if (chunkCounter != 0 && lastChunk.buf.position() == 0) {
+            logger.trace("The last chunk received from ReadableStreamChannel has no data, discarding");
```

Comment:
last PutChunk in building state having no data is the mark of last chunk or chunks from the `ReadableStreamChannel` having no data. I could rephrase the message to make it more general.